### DISCUSSION TITLE
Is there a way embed Process' Window on linux

### DISCUSSION THREAD

**Discussion by fatihbahceci** [AvaloniaUI/Avalonia#9690] [Q&A]
Hello as you know the VLCPlayer not supporting RTSP stream on debian bullseye and pardus21 anymore. 
I'm searching a way put native window to Control on linux for Avalonia. 

On windows it's so easy. 
```

        [DllImport("user32.dll", SetLastError = true)]
        private static extern bool SetWindowPos(IntPtr hWnd, int X, int Y, int cx, int cy, uint flags);

        [DllImport("user32.dll")]
        private static extern IntPtr SetParent(IntPtr hWndChild, IntPtr hWndNewParent);
...
 if (ffProcess.MainWindowHandle != IntPtr.Zero) {
             SetParent(ffProcess.MainWindowHandle, ctrl.Handle);
            MoveWindow(ffProcess.MainWindowHandle, 0, 0, ctrl.Width, ctrl.Height, true);
}
```

is there a way embedding process' window  for linux?

**Comment by maxkatz6** [comment]
You can use NativeControlHost.
See
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/NativeEmbedPage.xaml.cs#L56 https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.NetCore/NativeControls/Gtk/EmbedSample.Gtk.cs

Though, control embedding on Linux has its limitations https://github.com/AvaloniaUI/Avalonia/issues/6124

**Reply by fatihbahceci** [reply]
Thanks for the reply. I'll work on it. Hope I can handle it. If success I'll accept the answer and put here the final code.

**Reply by fatihbahceci** [reply]
Hello me again. In this [sample](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.NetCore/NativeControls/Gtk/EmbedSample.Gtk.cs) the  program sends existing control handle to mplayer as parameter. 

```
 _mplayer = Process.Start(new ProcessStartInfo("mplayer",
            $"-vo x11 -zoom -loop 0 -wid {control.Handle.ToInt64()} \"{nodes}\"")
        {
            UseShellExecute = false,

        });
```
but in my case I have to wait window's handle number from the Process and update NativeControl with this handle number. 
Do you have any advice about this. 

```
if (ffProcess.MainWindowHandle != IntPtr.Zero) { ... 
```


Thanks again.

**Reply by kekekeks** [reply]
`Process.MainWindowHandle` is not supported on non-windows platforms.
